File: backend/src/main/java/olihef/stratvis/sessions/Session.java
package olihef.stratvis.sessions;

import java.time.Instant;

public class Session {
	private final int sessionId;
	private final Instant creationTime;
	private Snapshot snapshot;

	public Session(int sessionId, Instant creationTime, Snapshot snapshot) {
		this.sessionId = sessionId;
		this.snapshot = snapshot;
		this.creationTime = creationTime;
	}

	public int sessionId() {
		return sessionId;
	}

	public Instant creationTime() {
		return creationTime;
	}

	public Snapshot snapshot() {
		return snapshot;
	}

	public void snapshot(Snapshot snapshot) {
		this.snapshot = snapshot;
	}
}

File: frontend/src/types/Session.ts
import type { AnalysisOutput } from '@stratvis/contracts';

export type AnalysisPoiWithMap = AnalysisOutput['points_of_interest'][number] & {
	lng?: number;
	lat?: number;
	range_m?: number;
};

export type AnalysisWithMap = Omit<AnalysisOutput, 'points_of_interest'> & {
	points_of_interest: AnalysisPoiWithMap[];
};

export interface Snapshot {
	label: string;
	minLng: number;
	minLat: number;
	maxLng: number;
	maxLat: number;
	usedZoom: number;
	analysis: AnalysisWithMap | null;
	creationTime: string;
}

export default interface Session {
	sessionId: number;
	snapshot: Snapshot;
	creationTime: string;
}
